pkg/framework/deployment.go

```diff
@@ -43,6 +44,17 @@ func DeleteDeployment(c client.Client, deployment *kappsapi.Deployment) error {
 	})
 }
 
+// UpdateDeployment updates the specified deployment
+func UpdateDeployment(c client.Client, deployment *kappsapi.Deployment, updated *kappsapi.Deployment) error {
```

Comment:
Does this work? I assume the only reason a patch would fail is because there is a conflict, in which case you'd just be retrying the same operation over and over and constantly get conflicts no?